Sprint_5

ezviagin opened the pull request:
###  UI тестирование сайта Stellar Burgers
Реализованы:
* class Credentials 
* class Account
* class StellarBurgerLocators
* UI-тесты на проверку:
  - регистрации
  - авторизации через различные страницы
  - выход из аккаунта
  - переходы по основным элементам главной страницы
  - переходы по конструктору ингредиентов

ezviagin committed:
UI tests for StellarBurgers website

AlexKitIt commented on locators.py:
```diff
@@ -0,0 +1,65 @@
+        PASSWORD_FIELD = (By.XPATH, ".//label[text()='Пароль']/..//input")                          # поле "Пароль"
+
+        # Getters
+        @staticmethod
+        def get_email_field_value(email: str) -> tuple[str, str]:
+            return (By.XPATH, "//input[@type='text' and @value='" + email + "']")
+
+        @staticmethod
+        def get_password_field_value(password: str) -> tuple[str, str]:
+            return (By.XPATH, "//input[@type='password' and @value='" + password + "']")
```
Тут чем не удружили простые переменные?

ezviagin replied:
@AlexKitIt функции получают введенное значение value через локатор, затем ждём, что поле заполнили. У меня была проблема - через раз в случайном порядке валились тесты, так как не успевали заполниться эти поля, получение значение value и ожидание visibility_of_element_located проблему решило.

Подскажи пожалуйста, как лучше сделать в таком случае?

AlexKitIt replied:
Еще ни разу не встречалось ошибки, что не "не успевали заполниться эти поля" selenium не переходит к следующему шагу пока не выполнит предыдущий, send_keys отрабатывает не одномоментной вставкой а посимвольным вводом, поэтому пока все символы не будут переданы в input следующий шаг не запустится. Проблема была явно не в этом

ezviagin replied:
@AlexKitIt да, удалив этот код, тоже убедился, что не в этом была проблема. Всё понял, огромное спасибо тебе!

AlexKitIt commented on tests/test_account.py:
```diff
@@ -0,0 +1,39 @@
+
+
+class TestAccount:
+    @staticmethod
```
Нужно исправить: все методы внутри тестового класса не статичные

ezviagin replied:
Исправлено

AlexKitIt commented on urls.py:
```diff
@@ -0,0 +1,5 @@
+URL_BASE = "https://stellarburgers.nomoreparties.site/"
+URL_LOGIN = "login"
+URL_ACCOUNT = "account"
```
Эти урлы почему не такие как URL_PROFILE?

ezviagin replied:
Исправлено

AlexKitIt commented on utils.py:
```diff
@@ -0,0 +1,22 @@
+
+
+
+def get_email():
+    username_length = random.randint(6, 12)
+    username = ''.join(random.choices(string.ascii_lowercase + string.digits, k=username_length))
+
+    number = random.randrange(1000, 1999)
+
+    domains = ['mail.ru', 'yahoo.com', 'hotmail.com', 'outlook.com', 'yandex.ru', 'ya.ru', 'outlook.com', 'gmail.com']
+    domain = random.choice(domains)
+
+    return f"{username}{str(number)}@{domain}"
+
+
+def get_password():
+    # Generate a random password of length 6-12
+    return ''.join(random.choices(string.ascii_letters + string.digits, k=random.randint(6, 64)))
```
Нужно исправить: эти вспомогательные методы живут в helpers

ezviagin replied:
Исправлено

ezviagin committed:
Fixes based on the notes:

- do not use static methods in test classes
- mode helper functions to helper.py
- remove value text check in __fill_credentials() function, add clear() before send_keys()
- refactored urls constant values

ezviagin committed:
Fixes based on the notes:

- do not use static methods in test classes
- mode helper functions to helper.py
- remove value text check in __fill_credentials() function, add clear() before send_keys()
- refactored urls constant values

ezviagin committed:
Merge branch 'develop' of https://github.com/ezviagin/Sprint_5 into develop

ezviagin committed:
Merge branch 'develop' of https://github.com/ezviagin/Sprint_5 into develop

ezviagin committed:
Merge branch 'develop' of https://github.com/ezviagin/Sprint_5 into develop
